app/Http/Controllers/AuthController.php

```diff
@@ -83,67 +139,21 @@ public function install(Request $request)
         }
     }
 
-    public function load(Request $request)
-    {
+    /**
+     * App uninstall callback url as defined in app configuration
+     */
+    public function uninstall(Request $request, BigCommerceService $bigCommerceService) {
         $this->validate($request, ['signed_payload' => 'required']);
         $signedPayload = $request->input('signed_payload');
 
-        $verifiedSignedRequestData = $this->verifySignedRequest($signedPayload);
-        if (!$verifiedSignedRequestData || !isset($verifiedSignedRequestData['context'])) {
-            return new Response('Error: The signed request from BigCommerce could not be validated', 400);
-        } else {
-            $store = Store::where('context', $verifiedSignedRequestData['context'])->first();
-            if (!$store) {
-                return new Response('Error: Store could not be found', 400);
-            }
-
-            $this->storeToSession([
-                'access_token' => $store['access_token'],
-                'context' => $verifiedSignedRequestData['context'],
-                'store_hash' => $verifiedSignedRequestData['store_hash']
-            ]);
-
-            $store_id = $store['id'];
-            $configRow = Config::where('store_id', $store_id)->first();
-
-            $viewData = [];
-            if (isset($configRow['id'])) {
-                $config = Config::find($configRow['id']);
-                if ($config->active > 0) {
-                    $activeStatus = $config->active;
-                } else {
-                    $activeStatus = null;
-                }
-                $viewData = [
-                    'shopkey' => $config->shopkey,
-                    'active_status' => $activeStatus
-                ];
-            }
-
-            return view('app', $viewData);
+        $data = $bigCommerceService->verifySignedRequest($signedPayload);
+        if ($data['user']['id'] != $data['owner']['id']) {
```

Comment:
The Controller is not responsible for authentication, that should be done by a [`Middleware`](https://laravel.com/docs/8.x/middleware).